File: tournaments/differentValuesInMultiplicationTable/differentValuesInMultiplicationTable.cpp
int differentValuesInMultiplicationTable(int n, int m) {
  int result = 0;
  for (int value = 1; value <= n * m; value++) {
    for (int i = 1; i <= std::min(n, m); i++) {
      if (std::max(n, m) >= value / i && value % i == 0) {
        result++;
        break;
      }
    }
  }
  return result;
}

File: tournaments/differentValuesInMultiplicationTable/differentValuesInMultiplicationTable.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub differentValuesInMultiplicationTable {
    my ($n, $m) = @_;

    use List::Util qw[min max];

    my $result = 0;

    for (my $i = 1; $n * $m >= $i; ++$i) {
        for (my $j = 1; min($n, $m) >= $j; ++$j) {
            if (max($n, $m) >= $i / $j && !($i % $j)) {
                ++$result;
                last;
            }
        }
    }

    return $result;
}
